eclipse-theia/theia
# [filesystem] Fix missing "out of sync error"

which should be shown if one try to save pending editor changes to a file which was modified in background.

Fixes eclipse-theia/theia#7102

Thanks @akosyakov to provide this fix!

#### How to test
- do some changes to a file
- modify the opened file on terminal
- try to save the dirty editor
  - it should warn you:
    <img width="498" alt="Screen Shot 2020-02-12 at 11 20 34" src="https://user-images.githubusercontent.com/914497/74326981-a3ac6a80-4d8b-11ea-9da4-7d9806137544.png">
- test closing with dirty and not against the file modified in the background and not
- test typing fast and slow with auto save
- test reloading a page with a file deleted in the meantime
- test file move and delete with dirty and not editors

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
